rpc/grpc/src/main/java/com/spotify/heroic/rpc/grpc/GrpcRpcProtocolServer.java

```diff
@@ -103,46 +102,61 @@ public GrpcRpcProtocolServer(
     private GrpcRpcContainer setupContainer() {
         final GrpcRpcContainer container = new GrpcRpcContainer();
 
-        container.register(GrpcRpcProtocol.METADATA, empty -> async.resolved(localMetadata));
+        container.register(GrpcRpcProtocol.METADATA, empty -> {
+            return async.resolved(metadataProvider.getMetadata());
+        });
 
         container.register(GrpcRpcProtocol.METRICS_FULL_QUERY,
-            g -> g.apply(metrics, MetricBackendGroup::query));
+            g -> g.apply(metrics, MetricBackendGroup::query)
+        );
```

Comment:
This whitespace change and the ones below are cancelled in the next commit. Would be cleaner if the change is never introduced.